### DISCUSSION TITLE
Server for DiaNN

### DISCUSSION THREAD

**Discussion by HarantK** [vdemichev/DiaNN#478] [Q&A]
Dear Dr. Demichev
We are running Dia-NN on a virtual windows machine on our server. Now we have the possibility to buy new hardware and migrate to new server - 120 cores and the corresponding amount of memory. Is there any specification for new hardware for optimal performance - processor architecture, memory amount, type of storage, and storage amount?   Our IT department asks me for software requirements with emphasis on server usage.
With regards
Karel Harant

**Comment by vdemichev** [comment]
Hi Karel,

Some users have reported that scaling beyond 32-40 cores does not work well I think. So having more cores only beneficial in the sense that you can run more DIA-NN analyses in parallel. In that case though having multiple 'desktop-type consumer-grade' PCs is more cost-effective. 

The performance is roughly number of cores * their operating frequency - on long lib-free analyses. With a spectral library, there's some bias towards having faster cores / faster access to disk. Local storage is highly beneficial, loading from networks will be a bottleneck in our experience.

RAM should be 128GB+ for lib-free (although most analyses are fine with 32GB), 64Gb+ for library-based.

Best,
Vadim
